types/defines/event-subscriptions.d.ts

```diff
@@ -0,0 +1,80 @@
+type EventSubscriptionEvent<
+	Service extends string,
+	Type extends string,
+	Source,
+	Payload = {},
+> = {
+	type: `cf.${Service}.${Type}`;
+	source: { service: Service } & Source;
+	payload: Payload;
+	metadata: {
+		accountId: string;
+		eventId: string;
+		eventSubscriptionId: string;
+		eventTimestamp: string; // RFC3339 timestamp
+		eventSchemaVersion: 1;
+	};
+};
+
+// Super Slurper
+export type EventMessageSuperSlurper =
+	| EventSubscriptionEvent<"superSlurper", "jobStarted", { jobId: string }>
+	| EventSubscriptionEvent<"superSlurper", "jobPaused", { jobId: string }>
+	| EventSubscriptionEvent<"superSlurper", "jobResumed", { jobId: string }>
+	| EventSubscriptionEvent<
+			"superSlurper",
+			"jobFinished",
+			{ jobId: string },
+			{
+				totalObjectsCount: number;
+				migratedObjectsCount: number;
+				failedObjectsCount: number;
+			}
+	  >
+	| EventSubscriptionEvent<"superSlurper", "jobAborted", { jobId: string }>;
+
+// Workflows
+type SubscriptionEventSourceWorkflows = {
+	workflowId: string;
+	workflowName: string;
+	versionId: string;
+	instanceId: string;
+};
```

Comment:
It's a bit weird that workflows has a helper for this, but superSlurper doesn't. Since this type will be exposed to users, I wonder if we should inline it?